packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -477,7 +477,7 @@ class _BuildInstance {
     if (results.any((bool result) => !result)) {
       return false;
     }
-    final AsyncMemoizer<bool> memoizer = pending[node.target.name] ??= AsyncMemoizer<bool>();
+    final AsyncMemoizer<bool> memoizer = (pending[node.target.name] ??= AsyncMemoizer<bool>()) as AsyncMemoizer<bool>;
```

Comment:
This is a mistake in the type of pending above, it should take an `AsyncMemoizer<bool>` so it doesn't require a cast. Not necessary to fix in this PR, but otherwise please file an issue and assign to me

Reply:
Fixed in the upcoming commit (rebase on master to remove conflicts)